tests/code_wars/test_subtract_the_sums.py

```diff
@@ -1,132 +1,143 @@
-# sum = 3+1+5 = 9
-# n = 315-9 = 306 (not in the list)
-# sum = 3+0+6 = 9
-# n =306-9 = 297 (not in the list)
-# .
-# .
-# .
-# ...until you find the first n in the list below.
-# 1-kiwi
-# 2-pear
-# 3-kiwi
-# 4-banana
-# 5-melon
-# 6-banana
-# 7-melon
-# 8-pineapple
-# 9-apple
-# 10-pineapple
-# 11-cucumber
-# 12-pineapple
-# 13-cucumber
-# 14-orange
-# 15-grape
-# 16-orange
-# 17-grape
-# 18-apple
-# 19-grape
-# 20-cherry
-# 21-pear
-# 22-cherry
-# 23-pear
-# 24-kiwi
-# 25-banana
-# 26-kiwi
-# 27-apple
-# 28-melon
-# 29-banana
-# 30-melon
-# 31-pineapple
-# 32-melon
-# 33-pineapple
-# 34-cucumber
-# 35-orange
-# 36-apple
-# 37-orange
-# 38-grape
-# 39-orange
-# 40-grape
-# 41-cherry
-# 42-pear
-# 43-cherry
-# 44-pear
-# 45-apple
-# 46-pear
-# 47-kiwi
-# 48-banana
-# 49-kiwi
-# 50-banana
-# 51-melon
-# 52-pineapple
-# 53-melon
-# 54-apple
-# 55-cucumber
-# 56-pineapple
-# 57-cucumber
-# 58-orange
-# 59-cucumber
-# 60-orange
-# 61-grape
-# 62-cherry
-# 63-apple
-# 64-cherry
-# 65-pear
-# 66-cherry
-# 67-pear
-# 68-kiwi
-# 69-pear
-# 70-kiwi
-# 71-banana
-# 72-apple
-# 73-banana
-# 74-melon
-# 75-pineapple
-# 76-melon
-# 77-pineapple
-# 78-cucumber
-# 79-pineapple
-# 80-cucumber
-# 81-apple
-# 82-grape
-# 83-orange
-# 84-grape
-# 85-cherry
-# 86-grape
-# 87-cherry
-# 88-pear
-# 89-cherry
-# 90-apple
-# 91-kiwi
-# 92-banana
-# 93-kiwi
-# 94-banana
-# 95-melon
-# 96-banana
-# 97-melon
-# 98-pineapple
-# 99-apple
-# 100-pineapple
-starting_number = 325
-def get_sum_of_digits in str(starting_number):
+# did not perform a true test because it is too time consuming to do the manual
+# math to create a answer key to check the fucntion against. Otherwise, I would
+# be using the function to test itself - not ideal
 
-    sum_of_digits = 0
-    for starting_number in str(n)
-        sum_of_digits += int(starting_number)
-    starting_number_minus_sum = starting_number - sum_of_digits
-    return starting_number_minus_sum
+# Issue reported with Code Wars - no solution possible with actual attempt
+# returns ""returned 'pineapple' and should be 'apple'
+# even with changing every pineapple to apple one by one in my answer key
+# it still returns same error
+
+import random
+from code_challenges.code_wars.subtract_the_sums import subtract_sum
+
+# fruit as a dictionary with fruit as key and list of numbers as values
+# fruit_dict = {
+#     "apple": [99, 90, 81, 72, 63, 54, 45, 36, 27, 18, 9],
+#     "banana": [96, 94, 92, 73, 71, 50, 48, 29, 25, 6, 4],
+#     "cherry": [89, 87, 85, 66, 64, 62, 43, 41, 22, 20],
+#     "cucumber": [80, 78, 59, 57, 55, 34, 13, 11],
+#     "grape": [86, 84, 82, 61, 40, 38, 19, 17, 15],
+#     "kiwi": [93, 91, 70, 68, 49, 47, 26, 24, 3, 1],
+#     "melon": [97, 95, 76, 74, 53, 51, 32, 30, 28, 7, 5],
+#     "orange": [83, 60, 58, 39, 37, 35, 16, 14],
+#     "pear": [88, 69, 67, 65, 46, 44, 42, 23, 21, 2],
+#     "pineapple": [100, 98, 79, 77, 75, 56, 52, 33, 31, 12, 10, 8],
+# }
+
+
+fruit_dict = [
```

Comment:
I missed this I think due to trying to get the actually commits sorted out. 

 - The commented out fruit_dict is no longer relevant to this solution so should be removed.  
 - The `fruit_dict` that is used isn't a dict.  It's pretty common when getting started to name the variables after the Python type,  like book_list or counts_dict.  This really should be done, if you really need to in order to keep track of what is what, okay, but know that it isn't conventional.   Better names would be book_list --> books and counts_dict --> letter_counts or letter_to_counts. So in this case just `fruits`
 - `fruit_dict` on line 30 is in the `global` scope and is called a `global` varable.  It's available for use anywhere in this file.  Global variable by convention are UPPER_CASE so this should be FRUIT_DICT or rather FRUITS = ..., but anytime a variable is put in the global scope, you have to ask is it needed there.  For example is it used in many functions within the file? Is it used across files? There are probably a few other cases, but in this case no, so it shouldn't be global to begin with but rather housed within the function.

Reply:
Create a new clean up branch, and make these fixes and PR for them.